### PR TITLE
cloak-pt: init at 2.9.0

### PR DESCRIPTION
**Pull request by BANanaD3V**
## Description of changes

Packaged cloak. This package provides cloak client (ck-client binary) and server (ck-server binary).
The homepage is https://github.com/cbeuw/Cloak, but I use my fork because the homepage got bad go.mod and doesn't build on nix.
Cloak is a pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule {
+  pname = "Cloak";
+  version = "2.8.0";
+
+  src = fetchFromGitHub {
+    owner = "BANanaD3V"; # Actual upstream (cbeuw) got a go.mod issue so we use this
```
Comment (sikmir) [inline pkgs/by-name/cl/cloak-pt/package.nix:11]: Please use upstream repo with a proper patch if required.

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    rev = "6b08af0";
+    hash = "sha256-2nk/sz2pNVt7rM+6omj+FK1U039swFqjHmnE4n3YC4E=";
+  };
+
+  vendorHash = "sha256-xyzFsk1frnj+HbYpXTjrZzD1NM5mokNv6maCA4CsT/w=";
+
+  doCheck = false;
+
+  ldflags = [
+    "-X main.version=master(6b08af0)"
```
Comment (sikmir) [inline pkgs/by-name/cl/cloak-pt/package.nix:22]:
```suggestion
    "-X main.version=${version}"
```

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule {
```
Comment (sikmir) [inline pkgs/by-name/cl/cloak-pt/package.nix:6]:
```suggestion
buildGoModule rec {
```

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  vendorHash = "sha256-xyzFsk1frnj+HbYpXTjrZzD1NM5mokNv6maCA4CsT/w=";
+
+  doCheck = false;
+
+  ldflags = [
+    "-X main.version=master(6b08af0)"
+  ];
+  meta = with lib; {
+    homepage = "https://github.com/cbeuw/Cloak/";
+    description = "Cloak is a pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination.";
```
Comment (sikmir) [inline pkgs/by-name/cl/cloak-pt/package.nix:26]:
```suggestion
    description = "Cloak is a pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination";
```

File: pkgs/by-name/cl/cloak-pt/package.nix (hunk 1 above)
Comment (BANanaD3V) [reply pkgs/by-name/cl/cloak-pt/package.nix:11]: Sure. We got some discussion going in the PR i sent to the upstream already.

File: pkgs/by-name/cl/cloak-pt/package.nix (hunk 2 above)
Comment (IogaMaster) [reply pkgs/by-name/cl/cloak-pt/package.nix:22]: +1

**Review by IogaMaster — CHANGES_REQUESTED** [review]
Please Squash all of the commits.

You should be left with:

- `maintainers: add bananad3v`
- `cloak-pt: init at 2.8.0`

**Comment by BANanaD3V** [comment]
> Please Squash all of the commits.
> 
> You should be left with:
> 
>     * `maintainers: add bananad3v`
> 
>     * `cloak-pt: init at 2.8.0`

If the cloak upstream dev doesn't tell me anything the pr in the near time i will do it and add a patch it seems. Yeah

File: pkgs/by-name/cl/cloak-pt/package.nix (hunk 4 above)
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:26]: The description should not start with the package name

See https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  vendorHash = "sha256-xyzFsk1frnj+HbYpXTjrZzD1NM5mokNv6maCA4CsT/w=";
+
+  doCheck = false;
+
+  ldflags = [
+    "-X main.version=master(6b08af0)"
+  ];
+  meta = with lib; {
```
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:24]:
```suggestion
  meta = {
```
Related to #292468

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    homepage = "https://github.com/cbeuw/Cloak/";
+    description = "Cloak is a pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination.";
+    longDescription = ''
+      Cloak is not a standalone proxy program. Rather, it works by masquerading proxied traffic as normal web browsing activities. In contrast to tra>
+
+      To any third party observer, a host running Cloak server is indistinguishable from an innocent web server. Both while passively observing traff>
+    '';
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ bananad3v ];
```
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:33]:
```suggestion
    license = lib.licenses.gpl3;
    maintainers = with lib.maintainers; [ bananad3v ];
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -2151,6 +2151,12 @@
     githubId = 18467667;
     name = "Alexander Bantyev";
   };
+  bananad3v = {
+    email = "banana@banana.is-cool.dev";
+    github = "bananad3v";
```
Comment (yunfachi) [inline maintainers/maintainer-list.nix:2156]:
```suggestion
    github = "BANanaD3V";
```
https://github.com/NixOS/nixpkgs/blob/5164ebdf667c299c4e688918b87d8606c0b49707/maintainers/maintainer-list.nix#L25

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  ldflags = [
+    "-X main.version=master(6b08af0)"
+  ];
+  meta = with lib; {
+    homepage = "https://github.com/cbeuw/Cloak/";
+    description = "Cloak is a pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination.";
+    longDescription = ''
+      Cloak is not a standalone proxy program. Rather, it works by masquerading proxied traffic as normal web browsing activities. In contrast to tra>
+
+      To any third party observer, a host running Cloak server is indistinguishable from an innocent web server. Both while passively observing traff>
```
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:30]: Why are the lines truncated with the `>`?

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    '';
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ bananad3v ];
+  };
+}
+
+
+
+
```
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:39]: Useless empty lines. You can use [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) for convenience

File: pkgs/by-name/cl/cloak-pt/package.nix (hunk 1 above)
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:11]: https://github.com/cbeuw/Cloak/pull/256 was merged, so you can use upstream

**Review by yunfachi — CHANGES_REQUESTED** [review]

File: pkgs/by-name/cl/cloak-pt/package.nix (hunk 4 above)
Comment (AndersonTorres) [inline pkgs/by-name/cl/cloak-pt/package.nix:26]:
```suggestion
    description = "Pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination";
```

**Review by IogaMaster — APPROVED** [review]

File: pkgs/by-name/cl/cloak-pt/package.nix [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    vendorHash = "sha256-kkb/gPnDbJvfc5Qqc5HuM1c9OwOu1ijfO7nNNnY3mOo=";
+
+    doCheck = false;
+
+    ldflags = [
+      "-X main.version=master(6b08af0)"
+    ];
+    meta = with lib; {
+      homepage = "https://github.com/cbeuw/Cloak/";
+      description = "Pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination.";
```
Comment (yunfachi) [inline pkgs/by-name/cl/cloak-pt/package.nix:28]:
```suggestion
      description = "Pluggable transport that enhances traditional proxy tools like OpenVPN to evade sophisticated censorship and data discrimination";
```
description should not end with punctuation marks

Comment (yunfachi) [reply pkgs/by-name/cl/cloak-pt/package.nix:28]: also don't forget the commit where you add yourself to the maintainers

**Comment by BANanaD3V** [comment]
Why is the check failing? It's formatted with nixfmt

**Comment by IogaMaster** [comment]
Two things:
Maintainer commit must be first
Run nixfmt, from the `nixfmt-rfc-style` package

**Comment by BANanaD3V** [comment]
> Two things: Maintainer commit must be first Run nixfmt, from the `nixfmt-rfc-style` package

done

**Review by yunfachi — APPROVED** [review]

**Comment by BANanaD3V** [comment]
@IogaMaster may we merge?

**Comment by IogaMaster** [comment]
I don't have merge rights, and I dont think merge bot does any but r-ryantm prs

**Comment by IogaMaster** [comment]
@NixOS/nixpkgs-merge-bot merge

**Pull request merged**